File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/signInCardParameters.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseCardParameters } from './baseCardParameters';

/**
 * Sharepoint sign in card view parameters
 */
export class SignInCardParameters extends BaseCardParameters {
    private primaryText: string;
    private description: string;
    private signInButtonText: string;

    /**
     * Initializes a new instance of the SignInCardParameters class
     */
    public SignInCardParameters() {
        // Do nothing
    }

    /**
     * Sets primary text property of type string
     */
    public set PrimaryText(primaryText: string) {
        this.primaryText = primaryText;
    }

    /**
     * Gets primary text property of type string
     */
    public get PrimaryText(): string {
        return this.primaryText;
    }

    /**
     * Sets description property of type string
     */
    public set Description(description: string) {
        this.description = description;
    }

    /**
     * Gets description property of type string
     */
    public get Description(): string {
        return this.description;
    }

    /**
     * Sets sign in button text property of type string
     */
    public set SignInButtonText(signInButtonText: string) {
        this.signInButtonText = signInButtonText;
    }

    /**
     * Gets sign in button text property of type string
     */
    public get SignInButtonText(): string {
        return this.signInButtonText;
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/SignInCardParameters.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint sign in card parameters.
    /// </summary>
    public class SignInCardParameters : BaseCardParameters
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="SignInCardParameters"/> class.
        /// </summary>
        public SignInCardParameters()
        {
            // Do nothing
        }

        /// <summary>
        /// Gets or Sets the primary text of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the primary text to display.</value>
        [JsonProperty(PropertyName = "primaryText")]
        public string PrimaryText { get; set; }

        /// <summary>
        /// Gets or Sets the description of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the description of the card view.</value>
        [JsonProperty(PropertyName = "description")]
        public string Description { get; set; }

        /// <summary>
        /// Gets or Sets the sign in button text of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the sign in button text of the card view.</value>
        [JsonProperty(PropertyName = "signInButtonText")]
        public string SignInButtonText { get; set; }
    }
}
